### PR TITLE
PP Robot Config -> core-swerve

### PR DESCRIPTION
**Pull request by hillelv**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/swerve/Swerve.java [hunk 1]
```diff
@@ -76,16 +75,24 @@ public SwerveStateHelper getStateHelper() {
...
-			constants.ppHolonomicDriveController(),
-			new RobotConfig(0, 0, new ModuleConfig(0, 0, 0, DCMotor.getFalcon500(1), 60, 2), 0),
-			() -> !Field.isFieldConventionAlliance(),
-			this
-		);//todo
+		RobotConfig robotConfig = PathPlannerUtils.DEFAULT_ROBOT_CONFIG;
+		try {
+			robotConfig = RobotConfig.fromGUISettings();
+		} catch (Exception e) {
+			new Alert(Alert.AlertType.ERROR, e.getMessage()).report();
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/swerve/Swerve.java:82]: add getLogPath + e.getMessage()

File: src/main/java/frc/robot/subsystems/swerve/Swerve.java [hunk 2]
```diff
@@ -76,16 +75,24 @@ public SwerveStateHelper getStateHelper() {
...
-			(speeds) -> driveByState(speeds, SwerveState.DEFAULT_PATH_PLANNER),
-			constants.ppHolonomicDriveController(),
-			new RobotConfig(0, 0, new ModuleConfig(0, 0, 0, DCMotor.getFalcon500(1), 60, 2), 0),
-			() -> !Field.isFieldConventionAlliance(),
-			this
-		);//todo
+		RobotConfig robotConfig = PathPlannerUtils.DEFAULT_ROBOT_CONFIG;
+		try {
+			robotConfig = RobotConfig.fromGUISettings();
+		} catch (Exception e) {
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/swerve/Swerve.java:81]:
```suggestion
		} catch (Exception exception){
```

File: src/main/java/frc/utils/auto/PathPlannerUtils.java [hunk 3]
```diff
@@ -27,6 +29,13 @@
 
 public class PathPlannerUtils {
 
+	public static final RobotConfig DEFAULT_ROBOT_CONFIG = new RobotConfig(
+		74,
+		8.6,
+		new ModuleConfig(0.048, 5.24, 0.96, DCMotor.getFalcon500(2), 60, 2),
+		0.577
+	);
+
```
Comment (YHerm) [inline src/main/java/frc/utils/auto/PathPlannerUtils.java:38]: Wouldn't it be very different for each robot?

Comment (hillelv) [reply src/main/java/frc/utils/auto/PathPlannerUtils.java:38]: Yes, I just put syncopa as default so we could still run pp if something went wrong with the gui. But it shouldn't happen, the config should always come from the gui

Comment (YHerm) [reply src/main/java/frc/utils/auto/PathPlannerUtils.java:38]: so maybe put in the config func in swerve a param for it, so people know that they need to enter it and change it for each robot

Comment (YHerm) [reply src/main/java/frc/utils/auto/PathPlannerUtils.java:38]: also are u sure about the number? especially the 2 in the   DCMotor.getFalcon500(2)

File: src/main/java/frc/utils/auto/PathPlannerUtils.java [hunk 4]
```diff
@@ -27,6 +29,13 @@
 
 public class PathPlannerUtils {
 
+	public static final RobotConfig DEFAULT_ROBOT_CONFIG = new RobotConfig(
```
Comment (YHerm) [inline src/main/java/frc/utils/auto/PathPlannerUtils.java:32]:
```suggestion
	public static final RobotConfig SYNCOPA_CONFIG = new RobotConfig(
```

File: src/main/java/frc/robot/subsystems/swerve/Swerve.java [hunk 5]
```diff
@@ -75,17 +74,24 @@ public SwerveStateHelper getStateHelper() {
...
-			new RobotConfig(0, 0, new ModuleConfig(0, 0, 0, DCMotor.getFalcon500(1), 60, 2), 0),
-			() -> !Field.isFieldConventionAlliance(),
-			this
-		);//todo
+	public void configPathPlanner(Supplier<Pose2d> currentPoseSupplier, Consumer<Pose2d> resetPoseConsumer, RobotConfig robotConfig) {
+		try {
+			robotConfig = RobotConfig.fromGUISettings();
+		} catch (Exception exception) {
+			new Alert(Alert.AlertType.ERROR, getLogPath() + exception.getMessage()).report();
+			exception.printStackTrace();
```
Comment (YHerm) [inline src/main/java/frc/robot/subsystems/swerve/Swerve.java:82]: you can do DriverStation.reportError()

**Pull request merged**